### DISCUSSION TITLE
sonoff rf bridge tasmota

### DISCUSSION THREAD

**Discussion by vizras** [arendst/Tasmota#11994] [Q&A] [answered]
Sonoff RF bridge Tasmota send commands after reconnect wifi or reboot.

**Comment by barbudor** [comment]
You probably have retained messages

Set Weblog 3 then reboot
Then post a copy of the web console logs
That should help identify the source of the command

NO SCREEN SHOT - USE FULL TEXT COPY
preferably use triple backticks to make the log readable

**Comment by vizras** [comment]
10:10:41.005 MQT: Attempting connection...
10:10:41.016 MQT: Connected
10:10:41.018 MQT: tele/bridge/LWT = Online (retained)
10:10:41.020 MQT: cmnd/bridge/POWER = 
10:10:41.021 MQT: Subscribe to cmnd/bridge/#
10:10:41.023 MQT: Subscribe to cmnd/tasmotas/#
10:10:41.024 MQT: Subscribe to cmnd/DVES_9EF66B_fb/#
10:10:41.026 MQT: Unsubscribe from homeassistant/status
10:10:41.029 MQT: tele/bridge/INFO1 = {"Info1":{"Module":"Sonoff Bridge","Version":"9.4.0(tasmota)","FallbackTopic":"cmnd/DVES_9EF66B_fb/","GroupTopic":"cmnd/tasmotas/"}}
10:10:41.031 MQT: tele/bridge/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"bridge-5739","IPAddress":"192.168.1.105"}}
10:10:41.034 MQT: tele/bridge/INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
10:10:41.255 SRC: MQTT
10:10:41.257 CMD: Group 0, Index 16, Command "RFKEY", Data "0"
10:10:41.268 MQT: stat/bridge/RESULT = {"RfKey16":"Learned sent"}
10:10:41.306 SRC: MQTT
10:10:41.308 CMD: Group 0, Index 15, Command "RFKEY", Data "0"
10:10:41.318 MQT: stat/bridge/RESULT = {"RfKey15":"Learned sent"}
10:10:41.739 DMP: A0 55
10:10:42.098 HTP: Console
10:10:44.537 QPC: Reset
10:10:45.443 MQT: tele/bridge/STATE = {"Time":"2021-05-03T10:10:45","Uptime":"0T00:00:09","UptimeSec":9,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Styrew","BSSId":"24:16:6D:EF:11:E0","Channel":11,"RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:03"}}
10:10:46.439 APP: Boot Count 104
10:10:46.650 CFG: Saved to flash at FB, Count 209, Bytes 4096
10:10:54.138 SRC: MQTT
10:10:54.140 CMD: Group 0, Index 16, Command "RFKEY", Data "0"

**Comment by barbudor** [comment]
Either someone is sending those RfKey commands over MQTT or the command has been   retained.
You need to purge your MQTT broker.

**Comment by vizras** [comment]
How to do it?

**Comment by barbudor** [comment, marked as answer]
What is your environment ? operating system ?

If you are using Windows, [MQTT-Explorer](http://mqtt-explorer.com/) is a nice tool and will show all topics on the broker. You will have to browse to find the retained ones and delete the with the trash icon.

If you have mosquitto, it comes with command line tools 
- `mosquitto_sub` to subscribe to your broker. you can subscribe to the topic `"cmnd/bridge/#"` and note what commands are listed as retained.
- `mosquitto_pub` can be used to publish an empty message to clear using the `-r` parameter

Check the doc of those tools

YOu may also have to understand why `Rfkey` commands have been published with retained flag

**Reply by sfromis** [reply]
MQTT-Explorer is also available for Linux and Mac. With `mosquitto_pub`, you may (depending on version) also need to have the `-n` (long form `--null-message`) parameter along with `-r` or `--retain`, and of course host and topic. The `mosquitto-clients` tools can also be installed separately without having Mosquitto itself.
